pkg/providers/imagefamily/bootstrap/parser_helpers.go

```diff
@@ -0,0 +1,684 @@
+	}
+
+	if s.VMVfsCachePressure != nil {
+		m["vm.vfs_cache_pressure"] = s.GetVMVfsCachePressure()
+	}
+
+	return base64.StdEncoding.EncodeToString([]byte(createSortedKeyValuePairs(m, "\n")))
+}
+
+func getShouldConfigContainerdUlimits(u *nbcontractv1.UlimitConfig) bool {
+	return u != nil
+}
+
+// getUlimitContent converts nbcontractv1.UlimitConfig to a string with key=value pairs.
+func getUlimitContent(u *nbcontractv1.UlimitConfig) string {
+	if u == nil {
+		return ""
+	}
+
+	header := "[Service]\n"
+	m := make(map[string]string)
+	if u.NoFile != nil {
+		m["LimitNOFILE"] = u.GetNoFile()
+	}
+
+	if u.MaxLockedMemory != nil {
+		m["LimitMEMLOCK"] = u.GetMaxLockedMemory()
+	}
+
+	return base64.StdEncoding.EncodeToString([]byte(header + createSortedKeyValuePairs(m, " ")))
+}
+
+// getPortRangeEndValue returns the end value of the port range where the input is in the format of "start end".
+func getPortRangeEndValue(portRange string) int {
+	if portRange == "" {
+		return -1
+	}
+
+	arr := strings.Split(portRange, " ")
+
+	// we are expecting only two values, start and end.
+	if len(arr) != MinArgs {
+		return -1
+	}
+
+	var start, end int
+	var err error
+
+	// the start value should be a valid port number.
+	if start, err = strconv.Atoi(arr[0]); err != nil {
+		log.Printf("error converting port range start value to int: %v", err)
+		return -1
+	}
+
+	// the end value should be a valid port number.
+	if end, err = strconv.Atoi(arr[1]); err != nil {
+		log.Printf("error converting port range end value to int: %v", err)
+		return -1
+	}
+
+	if start <= 0 || end <= 0 {
+		log.Printf("port range values should be greater than 0: %d", start)
+		return -1
+	}
+
+	if start >= end {
+		log.Printf("port range end value should be greater than the start value: %d >= %d", start, end)
+		return -1
+	}
+
+	return end
+}
+
+// createSortedKeyValuePairs creates a string with key=value pairs, sorted by key, with custom delimiter.
+func createSortedKeyValuePairs[T any](m map[string]T, delimiter string) string {
+	keys := []string{}
+	for key := range m {
+		keys = append(keys, key)
+	}
+
+	// we are sorting the keys for deterministic output for readability and testing.
+	sort.Strings(keys)
+	var buf bytes.Buffer
+	i := 0
+	for _, key := range keys {
+		i++
+		// set the last delimiter to empty string
+		if i == len(keys) {
+			delimiter = ""
+		}
+		buf.WriteString(fmt.Sprintf("%s=%v%s", key, m[key], delimiter))
+	}
+	return buf.String()
+}
+
+func getExcludeMasterFromStandardLB(lb *nbcontractv1.LoadBalancerConfig) bool {
+	if lb == nil || lb.ExcludeMasterFromStandardLoadBalancer == nil {
+		return true
+	}
+	return lb.GetExcludeMasterFromStandardLoadBalancer()
+}
+
+func getMaxLBRuleCount(lb *nbcontractv1.LoadBalancerConfig) int32 {
+	if lb == nil || lb.MaxLoadBalancerRuleCount == nil {
+		return int32(maxLBRuleCountDefault)
+	}
+	return lb.GetMaxLoadBalancerRuleCount()
+}
+
+func getGpuImageSha(vmSize string) string {
+	return common.GetAKSGPUImageSHA(vmSize)
+}
```

Comment:
why add a new getter?  I assume based on your funcmap pattern. 

There has to be a better way to do this than just wrapping tons of other functions in more functions.

Reply:
Adding @lilypan26  for comment on this particular one.
But you are right, it's because of the funcmap pattern. We follow the same funcMap pattern as AKS-RP does. We couldn't find a better one yet. If you happen to know one, please let us know.